app/models/miq_request.rb

```diff
@@ -237,8 +237,7 @@ def v_approved_by
   end
 
   def v_approved_by_email
-    emails = miq_approvals.inject([]) { |arr, a| arr << a.stamper.email unless a.stamper.nil? || a.stamper.email.nil?; arr }
-    emails.join(", ")
+    miq_approvals.collect { |i| i.stamper.try(:email) }.compact.join(", ")
```

Comment:
Nice, I can read what this method is doing now.  😉 

Please add a test for this.